compiler/compiler.cpp

```diff
@@ -212,14 +212,16 @@ bool compiler_execute(CompilerSettings *settings) {
   }
 
   SchedulerBase *scheduler;
-  if (G->settings().threads_count.get() == 1) {
+  const auto scheduler_threads = static_cast<int32_t>(G->settings().threads_count.get());
+  if (scheduler_threads == 1) {
     scheduler = new OneThreadScheduler();
   } else {
     auto s = new Scheduler();
-    s->set_threads_count(static_cast<int32_t>(G->settings().threads_count.get()));
+    s->set_threads_count(scheduler_threads);
     scheduler = s;
   }
 
+  vk::singleton<CppDestDirAsyncInitializer>::get().start(scheduler_threads + 1);
```

Comment:
When there is only 1 thread — for local development — let's do in synchronously in the same thread? For easier, one-thread debugging